### PR TITLE
Add ommer validation and pay rewards

### LINKED ISSUE #287: Implement ommer validation and ommer rewards
^

### PR DESCRIPTION
**Pull request by Bhargavasomu**
### What's new?
Ommer validation and rewards have been implemented

Closes #287

### How was it fixed?
Frontier yellow paper

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Comment by Bhargavasomu** [comment]
@SamWilsn @lightclient can you please review this whenever possible? Thanks a lot.

**Review by SamWilsn — APPROVED** [review]

**Comment by SamWilsn** [comment]
I renamed `add_ether` to `create_ether` and rebased. Lemme know if this looks good!

**Comment by Bhargavasomu** [comment]
> I renamed `add_ether` to `create_ether` and rebased. Lemme know if this looks good!

Looks good, thanks.

File: src/ethereum/frontier/state.py [hunk 1]
```diff
@@ -421,3 +421,23 @@ def write_code(sender: Account) -> None:
         sender.code = code
 
     modify_state(state, address, write_code)
+
+
+def create_ether(state: State, address: Address, amount: U256) -> None:
```
Comment (voith) [inline src/ethereum/frontier/state.py:426]: Can we rename this to `add_to_account_balance` instead?

I see a use case for this method in several places.
eg. https://github.com/ethereum/execution-specs/blob/04740bec742627f4214f88723cb8a091a48d8dcc/src/ethereum/frontier/spec.py#L421-L424
The code above can be replaced by `add_to_account_balance(state, sender, gas_refund_amount)`

File: src/ethereum/frontier/state.py [hunk 2]
```diff
@@ -421,3 +421,23 @@ def write_code(sender: Account) -> None:
...
+    """
+    Add newly created ether to an account.
```
Comment (voith) [inline src/ethereum/frontier/state.py:428]: We'll have to change this docstring if we rename this to `add_to_account_balance`

File: src/ethereum/frontier/state.py (hunk 1 above)
Comment (SamWilsn) [reply src/ethereum/frontier/state.py:426]: That's interesting. So while the transaction is executing, the ether used to pay for gas is "destroyed" (i.e. not in the origin account, and not in the coinbase account) and then "recreated"?

Comment (voith) [reply src/ethereum/frontier/state.py:426]: Let's just merge this and then I'll propose a change in a separate PR.

**Pull request merged**